toxcore/tox.h

```diff
@@ -686,6 +686,33 @@ struct Tox_Options {
      * Default: false. May become true in the future (0.3.0).
      */
     bool experimental_disable_dns;
+
+    /**
+     * @brief Whether the savedata data is owned by the Tox_Options object.
+     *
+     * If true, the setters for savedata and proxy_host try to copy the string.
+     * If that fails, the value is not copied and the member is set to the
+     * user-provided pointer. In that case, the user must not free the string
+     * until the Tox_Options object is freed. Client code can check whether
+     * allocation succeeded by comparing the value of the member to the
+     * user-provided pointer.
```

Comment:
I guess another option for error-handling would be to return a bool indicating if the set-call succeeded or not for the ones that might allocate? Feels more ergonomic to me, and would allow requiring the user to check that the set succeeded by nodiscarding the return value if/when the experimental option is made the default.

Reply:
Alright. I didn't want to change the function type, but done.